test/functional/combine_logs.py

```diff
@@ -53,6 +63,29 @@ def read_logs(tmp_dir):
 
     return heapq.merge(*[get_log_events(source, f) for source, f in files])
 
+
+def find_latest_test_dir():
+    """Returns the latest tmpfile test directory prefix."""
+    tmpdir = tempfile.gettempdir()
+
+    def join_tmp(basename):
+        return os.path.join(tmpdir, basename)
+
+    testdir_paths = [
+        join_tmp(name) for name in os.listdir(tmpdir) if
+        os.path.isdir(join_tmp(name)) and is_test_dir_name(name)
+    ]
+
+    testdir_paths.sort(key=lambda x: os.path.getmtime(x), reverse=True)
+
+    return testdir_paths[0] if testdir_paths else None
+
+
+def is_test_dir_name(path):
+    """Returns True if the filename indicates a test directory."""
+    return re.match(r'test\S{8}$', os.path.basename(path))
```

Comment:
Instead of matching just on "test*", which is used by other programs (such as our unit test runner, IIRC), couldn't you instead explicitly set a unique (or approximately unique) prefix for the temp folder name in the test framework?

Reply:
maybe match the silly unicode characters as well, they're *very* unlikely to arise anywhere else :)

Reply:
Oh, the unicode characters are only present when run through the test runner, in which case the tests are run in parallel, so this feature wouldn't turn out useful.

If the tests are run one-each, they use a different prefix, so I suggest to change that prefix.

Reply:
I've updated the test framework to share a `TMPDIR_PREFIX` constant which has a value of `"bitcoin_func_test_"`.